toucan_connectors/postgres/postgresql_connector.py

```diff
@@ -169,17 +175,40 @@ def describe(self, data_source: PostgresDataSource):
             res = cursor.description
         return {r.name: types.get(r.type_code) for r in res}
 
-    def get_model(self):
+    def get_model(self) -> Tuple[List[TableInfo], Dict]:
         """Retrieves the database tree structure using current connection"""
+        available_dbs = self._list_db_names()
+        databases_tree = []
+        for db in available_dbs:
+            with suppress(pgsql.OperationalError):
+                databases_tree += self._list_tables_info(db)
+        return DiscoverableConnector.format_db_model(databases_tree)
+
+    def get_model_with_info(self) -> Tuple[List[TableInfo], Dict]:
+        """Retrieves the database tree structure using current connection"""
+        available_dbs = self._list_db_names()
+        databases_tree = []
+        failed_databases = []
+        for db in available_dbs:
+            try:
+                databases_tree += self._list_tables_info(db)
+            except pgsql.OperationalError:
+                failed_databases.append(db)
+
+        tables_info = DiscoverableConnector.format_db_model(databases_tree)
+        metadata = dict()
+        if len(failed_databases):
+            metadata["info"] = {"Could not reach databases": failed_databases}
+        return (tables_info, metadata)
+
+    def _list_db_names(self):
         connection = pgsql.connect(**self.get_connection_params(database='postgres'))
         with connection.cursor() as cursor:
             cursor.execute("""select datname from pg_database where datistemplate = false;""")
-            available_dbs = [db_name for (db_name,) in cursor.fetchall()]
-            databases_tree = []
-            for db in available_dbs:
-                connection = pgsql.connect(**self.get_connection_params(database=db))
-                with connection.cursor() as cursor:
-                    cursor.execute(build_database_model_extraction_query())
-                    res = cursor.fetchall()
-                    databases_tree += res
-            return format_db_model(databases_tree)
+            return [db_name for (db_name,) in cursor.fetchall()]
+
+    def _list_tables_info(self, db):
```

Comment:
```suggestion
    def _list_tables_info(self, database_name: str) -> List[tuple]:
```
ou qch comme ca ptet plus explicite ? au début je pensais que `db` serait une instance d'une classe ORM ou chais pas quoi